tests/model_explainability/trustyai_service/trustyai_service_utils.py

```diff
@@ -0,0 +1,538 @@
+        data_path (str): Directory path containing data batch files.
+        trustyai_service (TrustyAIService): TrustyAIService that will register the model.
+        inference_service (InferenceService): Model to be registered by TrustyAI.
+        inference_config (dict[str, Any]): Inference config to be used when sending the inference.
+        inference_type (Optional[str]): Inference type to be used when sending the inference
+        protocol (Optional[str]): Protocol to be used when sending the inference
+    """
+    for root, _, files in os.walk(data_path):
+        for file_name in files:
+            file_path = os.path.join(root, file_name)
+
+            with open(file_path, "r") as file:
+                data = file.read()
+
+            current_observations = get_num_observations_from_trustyai_service(
+                client=client, token=token, trustyai_service=trustyai_service
+            )
+            expected_observations: int = current_observations + json.loads(data)[0]["shape"][0]
+
+            inference = UserInference(
+                inference_service=inference_service,
+                inference_config=inference_config,
+                inference_type=inference_type,
+                protocol=protocol,
+            )
+
+            res = inference.run_inference_flow(
+                model_name=inference_service.name,
+                inference_input=data,
+                use_default_query=False,
+                token=token,
+            )
+            LOGGER.debug(f"Inference response: {res}")
+
+            samples = TimeoutSampler(
+                wait_timeout=Timeout.TIMEOUT_5MIN,
+                sleep=1,
+                func=lambda: get_num_observations_from_trustyai_service(
+                    client=client, token=token, trustyai_service=trustyai_service
+                ),
+            )
+
+            for obs in samples:
+                if obs >= expected_observations:
+                    break
+            else:
+                raise AssertionError(f"Observations not updated. Current: {obs}, Expected: {expected_observations}")
+
+
+def wait_for_isvc_deployment_registered_by_trustyai_service(
+    client: DynamicClient, isvc: InferenceService, runtime_name: str
+) -> None:
+    """
+    Check if all the ModelMesh pods in a given namespace are
+    ready and have been registered by the TrustyAIService in that same namespace.
+
+    Args:
+        client (DynamicClient): The client instance for interacting with the cluster.
+        isvc (InferenceService): The InferenceService related to the deployment.
+        runtime_name (str): The name of the serving runtime of the isvc
+    """
+    label_selector = create_isvc_label_selector_str(isvc=isvc, resource_type="deployment", runtime_name=runtime_name)
+    trustyai_service = TrustyAIService(name=TRUSTYAI_SERVICE_NAME, namespace=isvc.namespace, ensure_exists=True)
+
+    def _check_deployments_ready() -> bool:
```

Comment:
use the sampler to get the deployments and then inside the for call to check replicas